File: dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

/**
 * i'm getting a string of numbers
 */
class ArmstrongNumbers {
  bool isArmstrongNumber(String number) {
    num count = 0;
    int numberLength = number.length;
    int armstrong = int.parse(number);

    List<String> str = number.split('');
    List<int> digits = str.map((digit) => int.parse(digit)).toList();

    for (int digit in digits) {
      count += pow(digit, numberLength);
    }

    print(count);

    return armstrong == count;
  }
}

File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include "math.h"
#include "stdlib.h"

bool is_armstrong_number(int candidate) {
  int length;
  int* digits = convertNumberIntoArray(number, &length);
  int sum = 0;
  for (int i = 0; i < length; i++) {
    sum += pow(digits[i], length);
  }

  free(digits);

  return sum == number;
}

int* convertNumberIntoArray(int number, int* length) {
  *length = (int)floor(log10((float)abs(number))) +1;
  int* arr = (int*) malloc(*length * sizeof(int));
  
  int i = 0;
  do {
    arr[i] = abs(number) % 10;
    number /= 10;
    i++;
  } while(number != 0);
  
  return arr;
}
